Backend/src/resolverMap.ts

```diff
@@ -112,7 +109,24 @@ const resolverMap: IResolvers = {
   },
 };
 
-export default resolverMap;
+/**
+ * Check if the person requesting something, is actually who they say they are.
+ * For example used for deletion. People may only delete themselves.
+ * @param args
+ * @param context
+ */
+const mayOnlyModifySelf = async (args: any, context: any) => {
+  const user = await userFacade.getUserByEmail(context.token.useremail);
+  if (!(user.username == args.username)) {
+    throw new ForbiddenError("You don't have permission to do that.");
+  }
+};
```

Comment:
I'm not as into this stuff as you are, but it struck me as odd that we call the userFacade here. Is this the intended use?

I am imagining this check to appear *during* an update, where it would already have access to token context (username) and the requested resource to change, which could then be compared - instead of manually calling the facade.

Reply:
Yeah. 
Issue is that only the email is on the token. 
So to get the username, to compare, I need to get the user first.

Reply:
Then another issue is, that not all users have usernames, do they? @HrBjarup

Reply:
I think they are forced now, either they choose a username or their username is their email - so no one should be "empty"

Reply:
I'm not sure. The Token is made here. Is the username known at that point? 
![image](https://user-images.githubusercontent.com/11192971/83298851-775f1180-a1f5-11ea-8dba-c743b923e3b0.png)

Reply:
The username is not necessarily known at that point. The user won't have a username the first time they log in with google. We could code it so that they have their username set to their email instantly the first time they log in, and still just present them with the choice of creating a new username.